unity/ComputerVision-UnityClient/Assets/Scripts/AppManager.cs

```diff
@@ -0,0 +1,370 @@
+        {
+            switch (NetworkManager.connectionMode)
+            {
+                case NetworkManager.ConnectionMode.WebSocket:
+                    throw new Exception("Not Implemented Yet");
+                default:
+                case NetworkManager.ConnectionMode.Rest:
+                    string uri = await networkManager.UriBasedOnConnectionMode();
+                    url = uri + urlSuffix;
+                    urlFound = true;
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// ImageSenderFlow Flow : Hide the GUI > CaptureScreenShot
+        ///  > ShowGUI > Shrink Image > Send Image to Server > Handle Server Response > Repeat
+        /// </summary>
+        IEnumerator ImageSenderFlow()
+        {
+            showGUI = false;
+            yield return new WaitForEndOfFrame();
+            int width = Screen.width;
+            int height = Screen.height;
+            Texture2D texture = new Texture2D(width, height, TextureFormat.RGB24, true);
+            texture.ReadPixels(new Rect(0, 0, width, height), 0, 0);
+            showGUI = serviceAlreadyStarted;
+            texture.Apply();
+            Debug.Log("ScreenShot taken");
+            byte[] imgBinary = ShrinkAndEncode(source: texture, targetWidth: serviceMode == ServiceMode.FaceRecognition ? 500 : 240);
+            Destroy(texture);
+            Debug.Log("Image has been shrunk and encoded, Sending Image to Server");
+            while (!urlFound && webRequestSent)
+            {
+                yield return null;
+            }
+            StartCoroutine(SendImageToServer(imgBinary));
+        }
+        
+        IEnumerator SendImageToServer(byte[] imgBinary)
+        {
+            webRequestSent = false;
+            WWWForm form = new WWWForm();
+            form.AddBinaryData("image", imgBinary);
+            UnityWebRequest www = UnityWebRequest.Post(url, form);
+            www.timeout = 5; // Timeout in seconds
+            yield return www.SendWebRequest();
+
+            // isHttpError True on response codes greater than or equal to 400.
+            // isNetworkError True on failure to resolve a DNS entry
+            if (www.isNetworkError || www.isHttpError)
+            {
+                Debug.Log("Error sending Image to SERVER");
+                Debug.Log(www.error);
+                if (www.responseCode == 503)
+                {
+                    Debug.Log("Training data update in progress, Sending another request in 2 seconds.");
+                    yield return new WaitForSeconds(2);
+                    StartCoroutine(SendImageToServer(imgBinary));
+                    yield break;
+                }
+                if (www.responseCode == 501)
+                {
+                    Debug.LogError("The server executing this call does not have GPU support.");
+                    yield return new WaitForSeconds(2);
+                    StartCoroutine(SendImageToServer(imgBinary));
```

Comment:
If the server does not GPU support, should we still be trying to SendImageToServer again?

Reply:
Good Catch, Thanks